torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -69,22 +69,20 @@ def decompress(fut):
 class OptimizerHookState(object):
     """
     Holds state for running optimizer in-line after DDP communication hook.
-    Currently contains optimizer class as well as optimizer args.
+    Currently contains only optimizer class which must have a method `step_param`.
     """
-    __slots__ = ['functional_optim_cls', 'functional_optim_args', 'functional_optimizer']
+    __slots__ = ['functional_optimizer']
 
-    def __init__(self, functional_optim_cls, *functional_optim_args):
-        # TODO: Support kwargs
-        self.functional_optim_cls = functional_optim_cls
-        self.functional_optim_args = functional_optim_args
-        self.functional_optimizer = self.functional_optim_cls(
+    def __init__(self, functional_optim_cls, *functional_optim_args, **functional_optim_kwargs):
+        self.functional_optimizer = functional_optim_cls(
             [],
-            *self.functional_optim_args,
+            *functional_optim_args,
```

Comment:
Optimizers usually only have kwargs. Do you have any example that needs args?

Reply:
AFAIK user is able to construct regular optimizer without specifying kwargs and just using the parameter order in constructor:
```
s = torch.optim.SGD(torch.nn.Linear(1,1).parameters(), 0.01)
```

so wanted to support the same behavior here.